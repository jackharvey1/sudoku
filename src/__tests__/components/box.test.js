import React from 'react';
import Box from '../../components/box';
import Renderer from 'react-test-renderer';

it('renders the Box component as intended', () => {
    const box = Renderer.create(<Box
        onClick={() => null}
        box={0}
        values={["1", "", "", "", "", "", "", "", ""]}
        lockedCells={[[0, 1]]}
        selectedBox={0}
        selectedSquare={0}
    />);
    expect(box.toJSON()).toMatchSnapshot();
});
