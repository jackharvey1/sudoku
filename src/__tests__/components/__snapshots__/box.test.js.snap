// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Box component as intended 1`] = `
<div
  className="css-vkhxgo"
>
  <div
    className="css-sjhpqe"
  >
    <Square
      box={0}
      isLocked={false}
      isSelected={true}
      onClick={[Function]}
      square={0}
      value="1"
    />
    <Square
      box={0}
      isLocked={true}
      isSelected={false}
      onClick={[Function]}
      square={1}
      value=""
    />
    <Square
      box={0}
      isLocked={false}
      isSelected={false}
      onClick={[Function]}
      square={2}
      value=""
    />
  </div>
  <div
    className="css-sjhpqe"
  >
    <Square
      box={0}
      isLocked={false}
      isSelected={false}
      onClick={[Function]}
      square={3}
      value=""
    />
    <Square
      box={0}
      isLocked={false}
      isSelected={false}
      onClick={[Function]}
      square={4}
      value=""
    />
    <Square
      box={0}
      isLocked={false}
      isSelected={false}
      onClick={[Function]}
      square={5}
      value=""
    />
  </div>
  <div
    className="css-sjhpqe"
  >
    <Square
      box={0}
      isLocked={false}
      isSelected={false}
      onClick={[Function]}
      square={6}
      value=""
    />
    <Square
      box={0}
      isLocked={false}
      isSelected={false}
      onClick={[Function]}
      square={7}
      value=""
    />
    <Square
      box={0}
      isLocked={false}
      isSelected={false}
      onClick={[Function]}
      square={8}
      value=""
    />
  </div>
</div>
`;
